python/spec/dependabot/python/update_checker/index_finder_spec.rb

```diff
@@ -309,5 +309,46 @@
         end
       end
     end
+
+    describe "#clean_check_and_remove_environment_variables" do
```

Comment:
Oh didn't spot that this is a private method. Could you move the `context "environment variable interpolation` block inside `describe "#index_urls` ?

I think you can then expect something like this instead

```ruby
context "environment variable interpolation" do
  let(:credentials) do
    [
      {
        "type" => "python_index",
        "index-url" => "https://${creds}@company.com/simple",
        "token" => "user:password",
        "replaces-base" => false
      },
      {
        "type" => "python_index",
        "index-url" => "https://${creds}@not.company.com/simple",
        "token" => "user:password",
        "replaces-base" => false
      }
    ]
  end

  it "cleans and removes the environment variables" do
    expect(index_urls).to match_array(
      [
        "https://user:password@company.com/simple",
        "https://user:password@not.company.com/simple"
      ]
    )
  end
end
```

Reply:
Thanks! I'll give it a go and see what failures I get. Thanks for your help with this 👍